aep/general/0005/aep.md

```diff
@@ -0,0 +1,84 @@
+# Designing an API
+
+This AEP serves as a high-level guide on designing an API that adheres to the
+AEPs.
+
+## Process summary
+
+1. enumerate the user case would like your API to satisfy.
+1. identify resources and add them to your API.
+1. identify standard operations and add them to your API.
+1. identify custom operations and add them to your API.
+
+## Enumerate use cases
+
+The first step in designing an API is understanding precisely what you would
+like your user to be able to accomplish your API.
+
+Enumerate one or more use cases to clarify what precisel
+
+For example:
+
+- Create a VM.
+- List all VMs owned by a company.
+- List all VMs owned by a user.
+- Restart a running VM.
+- Trade an in-game item with another player.
+
+Some best practices:
+
+- Attempt to define granular use cases that can be composed to
+  satisfy more complex use cases.
+
+## Identify resources
+
+Once your use cases are defined, consider how many of those can be accomplished
+by represented by an [API Resource Type][]: similar to object-oriented
+programming, these are use cases that can be represented by entities that are
+created, read, updated, and deleted.
```

Comment:
I agree there are parallels between OOP and ROAD, but this reads like the main similarity is CRUD operations, and CRUD isn't really an OOP principle.

Reply:
yeah agreed, re-reading this is a false equivalency. removed.